puppet.go

```diff
@@ -338,6 +342,13 @@ func (puppet *Puppet) SyncWithProfileOverride(override ProfileOverride) {
 	}
 	if len(override.PhotoURL) > 0 {
 		go puppet.backgroundAvatarUpdate(override.PhotoURL)
+	} else if len(override.PhotoMXC) > 0 {
+		mxc, err := id.ParseContentURI(override.PhotoMXC)
+		if err != nil {
+			puppet.log.Warnfln("Failed to parse mxc %s: %v", override.PhotoMXC, err)
+		} else {
+			go puppet.UpdateAvatarFromMXC(mxc)
```

Comment:
The other one runs in background because fetching can be slow, but this is just a request to the homeserver like `UpdateNameDirect`

```suggestion
			puppet.UpdateAvatarFromMXC(mxc)
```